### ISSUE TITLE
TLB's port of Automatic1111 is broken for Hypernetwork training since 2.1

### ISSUE THREAD

**Issue by DarkAlchy** [TheLastBen/fast-stable-diffusion#1062]
Run out of VRAM on a T4.  I have to --medvram to get it to work with a BS 1 BUT I get Loss: nan and it really isn't working.  If I leave the --medvram and add --no-half (supposedly that is the fix) I then load up the images and POW pytorch is out of memory.  This did work in 2.0 but two days later I tried HN again and it went to 2.1 to no longer work.

Was reading over on the Automatic1111 issues and someone else had a sort of similar issue where they said to come here because of the way Ben does stuff differently than they do.

**Comment by TheLastBen** [comment]
the problem is with V2.1 not with the colab

**Comment by DarkAlchy** [comment]
Is there any way I can go back to 2.0?

**Comment by TheLastBen** [comment]
yes, in the model download cell, use this in the "path to huggingface" box : stabilityai/stable-diffusion-2
make sure you insert your huggingface token above it

**Comment by DarkAlchy** [comment]
> yes, in the model download cell, use this in the "path to huggingface" box : `stabilityai/stable-diffusion-2`
> make sure you insert your huggingface token above it and choose v2.1-768 in the custom model version

I do not have that box, but here is what I do have:
![image](https://user-images.githubusercontent.com/10561690/208243227-b09682b3-ceed-44d3-8392-74985c92ceeb.png)

**Comment by TheLastBen** [comment]
click twice on the cell and change wherever you find stabilityai/stable-diffusion-2-1 to stabilityai/stable-diffusion-2

**Comment by DarkAlchy** [comment]
So, change code then?  

![image](https://user-images.githubusercontent.com/10561690/208243634-c04b9e63-6c75-4c2d-9f61-1eec8985c73f.png)

**Comment by DarkAlchy** [comment]
> click twice on the cell and change wherever you find `stabilityai/stable-diffusion-2-1` to `stabilityai/stable-diffusion-2`

It works but how is it that I am forced to use BS=1 (512x512) on 15.1 gig T4 whereas locally, doing the same thing, they can do BS=2 with 10gigs? 24gig is BS=19 so my 15 gig should be 2 to 19 (around BS of 12-14).

**Comment by nawnie** [comment]
Mid training or at start. I have no answers im going to try and replicate

**Comment by DarkAlchy** [comment]
> Mid training or at start. I have no answers im going to try and replicate

I got it to work but I used the techniques of 6GB card users where you do NOT do images every X amount of epochs as those take VRAM as well.  Seriously, there is something wrong with this colab code.

**Comment by TheLastBen** [comment]
> What did they do to 2.1 to cause this? Oh, and here is something else but the model used to train with has a completely different hash than the SAI one.

https://github.com/Stability-AI/stablediffusion/commit/c12d960d1ee4f9134c2516862ef991ec52d3f59e

**Comment by DarkAlchy** [comment]
> > What did they do to 2.1 to cause this? Oh, and here is something else but the model used to train with has a completely different hash than the SAI one.
> 
> [Stability-AI/stablediffusion@c12d960](https://github.com/Stability-AI/stablediffusion/commit/c12d960d1ee4f9134c2516862ef991ec52d3f59e)

No idea I am just saying that the model it is using on colab is not the same model we use locally.

**Comment by CJohnDesign** [comment]
If we train on 2.0, does the `.pt` file still work in 2.1?

**Comment by DarkAlchy** [comment]
> If we train on 2.0, does the `.pt` file still work in 2.1?

Sure does.
